language/move-prover/interpreter-testsuite/tests/concrete_check/global_basics.move

```diff
@@ -0,0 +1,79 @@
+module 0x2::A {
+    struct R has key {
+        f1: bool,
+        f2: u64,
+    }
+
+    #[test(a=@0x2)]
+    public fun move_to_ok(a: &signer) {
+        let r = R {f1: true, f2: 1};
+        move_to(a, r);
+    }
+
+    #[test(a=@0x2), expected_failure]
+    public fun move_to_err(a: &signer) {
+        let r1 = R {f1: true, f2: 1};
+        move_to(a, r1);
+
+        let r2 = R {f1: false, f2: 0};
+        move_to(a, r2);
+    }
+
+    #[test(a=@0x2)]
+    public fun move_from_ok(a: &signer): R acquires R {
+        let r = R {f1: true, f2: 1};
+        move_to(a, r);
+        let b = @0x2;
+        move_from<R>(b)
+    }
+
+    #[test, expected_failure]
+    public fun move_from_err(): R acquires R {
+        let b = @0x2;
+        move_from<R>(b)
+    }
+
+    #[test(a=@0x2)]
+    public fun borrow_imm_ok(a: &signer): u64 acquires R {
+        let r = R {f1: true, f2: 1};
+        move_to(a, r);
+        let b = @0x2;
+        borrow_global<R>(b).f2
+    }
+
+    #[test, expected_failure]
+    public fun borrow_imm_err(): u64 acquires R {
+        let b = @0x2;
+        borrow_global<R>(b).f2
+    }
+
+    #[test(a=@0x2)]
+    public fun borrow_mut_ok(a: &signer): u64 acquires R {
+        let r = R {f1: true, f2: 1};
+        move_to(a, r);
+        let b = @0x2;
+        *&mut borrow_global_mut<R>(b).f2 = 0;
+        borrow_global<R>(b).f2
+    }
+
+    #[test, expected_failure]
+    public fun borrow_mut_err(): u64 acquires R {
+        let b = @0x2;
+        borrow_global_mut<R>(b).f2
+    }
+
+    #[test(a=@0x2)]
+    public fun exists_true(a: &signer): bool {
+        let r = R {f1: true, f2: 1};
+        move_to(a, r);
+        let b = @0x2;
+        exists<R>(b)
```

Comment:
I'm wondering, do you want to assert on these exists checks?

Reply:
I guess either way works? The main goal of this testsuite is to check that the stackless VM behaves like the Move VM so I am "implicitly" relying on the return value of the function (and the cross-vm comparison) for checking.

Reply:
Oh, woah..that's really cool actually! I think leaving this as-is then is fine